.github/workflows/ci.yml

```diff
@@ -0,0 +1,34 @@
+name: CI
+
+on:
+  pull_request: {}
+  push:
+    branches:
+      - main
+
+jobs:
+  ci:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ATiltedTree/setup-rust@v1
+        with:
+          rust-version: stable
+          components: clippy
+
+      - name: Install dependencies
+        run: sudo apt-get install -y libpulse-dev libopus-dev libmp3lame-dev
+
+      - name: build
+        run: cargo build
+      - name: lint
+        run: cargo clippy -- -D warnings
+      - name: rustfmt
+        run: cargo fmt -- --check
+      - name: build
+        run: cargo check
```

Comment:
any reason to run cargo check again here after running build above?

Reply:
hahaha, i have no idea what that does. i only saw it in another CI recipe and blindly copied it. is it redundant? if so, i'll nuke it.

Reply:
ah! Yeah it basically does a `cargo build` minus any of the codegen. I guess people wanted this since Rusts compile times can be so slow 😢